modelerfour/modeler/modelerfour.ts

```diff
@@ -84,6 +127,62 @@ export class ModelerFour {
     });
     this.interpret = new Interpretations(session, this.codeModel);
 
+    this.preprocessOperations();
+  }
+
+  preprocessOperations() {
+    // preprocess to get all http operations flattend out into a nice neat collection
+    for (const { key: path, value: pathItem } of this.resolveDictionary(this.input.paths)) {
+      for (const httpMethod of [HttpMethod.Delete, HttpMethod.Get, HttpMethod.Head, HttpMethod.Options, HttpMethod.Patch, HttpMethod.Post, HttpMethod.Put, HttpMethod.Trace]) {
+        const httpOperation = pathItem[httpMethod];
+        if (httpOperation) {
+          this.inputOperations.push({
+            method: httpMethod,
+            path: this.interpret.getPath(pathItem, httpOperation, path),
+            pathItem,
+            operation: httpOperation
+          })
+        }
+      };
+    };
+  }
+
+  initApiVersionMode(apiVersionParameter: 'choice' | 'constant' | undefined, useModelNamespace: boolean | undefined) {
+    if (this.profileFilter.length > 0) {
+      // must be profile mode.
+      return 'profile';
+    }
+
+    // see how many api versions there are for all the operations
+    const allApiVersions = values(this.inputOperations).selectMany(each => {
+      const apiv = <Array<string>>this.interpret.xmsMetaFallback(each.operation, each.pathItem, 'apiVersions');
```

Comment:
Should the key at the end be `apiVersions` or `apiVersion`?

Reply:
It's `apiVersions` (in the xmsMetadata, we collect the apiversions for the node and toss them in an array)